resources-generator/src/main/kotlin/dev/icerock/gradle/generator/apple/AppleImagesGenerator.kt

```diff
@@ -60,16 +61,23 @@ class AppleImagesGenerator(
             val imagesContent = validFiles.joinToString(separator = ",\n") { file ->
                 val scale = file.scale
 
+                // language=js
+                val scaleString = if (file.svg) "" else ",\n\t\"scale\" : \"$scale\""
+
                 // language=js
                 return@joinToString """
                     {
                         "idiom" : "universal",
-                        "filename" : "${file.name}",
-                        "scale" : "$scale"
+                        "filename" : "${file.name}"$scaleString
                     }
                 """.trimIndent()
             }
 
+            val svgProperties = if (validFiles.any { file -> file.svg }) {
+                // language=js
+                ",\n\t\"properties\" : {\t\t\"preserves-vector-representation\" : true\n\t}"
+            } else ""
```

Comment:
maybe better to create mutableMap / JsonObject that will be filled in more understandable way. and after complete configuration - just map JsonObject to String

Reply:
I have replaced to a json builder